holopy/core/metadata.py

```diff
@@ -142,123 +45,112 @@ def __init__(self):
     def __str__(self):
         return ("Medium index not specified in Optics instance.")
 
```

Comment:
This is mostly unrelated to your PR, but these exception classes shouldn't be living in metadata. Actually, it looks like they aren't used anywhere in holopy so you should delete them (they are redundant with MissingParameter in any case)